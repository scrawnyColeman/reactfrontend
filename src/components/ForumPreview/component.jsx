import { useHistory } from 'react-router-dom';
import React from 'react';
import styled from 'styled-components';
import Button from '../Button/component';
import { colours } from '../../constants/styles';

const StyledContainer = styled.div`
    margin: 0 auto;
    width: 45vw;
    height: 600px;
`;
const StyledPostsContainer = styled.div`
    overflow-y: scroll;
    height: 580px;
`;
const StyledTitle = styled.div`
    padding: 6px;
    color: ${colours.secondary}
    text-align: left;
`;

const StyledWrapper = styled.div`
    display: grid;
    grid-template-rows: 1fr min-content;
    margin: 0 8px;
    text-align: left;
`;
const StyledHead = styled.div`
    display: flex;
    justify-content: space-between;
    color: ${colours.primary};
    font-size: 1.25rem;
`;
const StyledBody = styled.div`
    display flex;
    justify-content: space-between;
    color: ${colours.secondary}
`;

const ForumPreview = ({ onClick, posts }) => {
    const history = useHistory();
    return (
        <StyledContainer>
            <StyledTitle>Forum Posts:</StyledTitle>
            <StyledPostsContainer>
                {posts.map((post) => {
                    const { id, title, type, comments } = post;
                    return (
                        <StyledWrapper key={id}>
                            <StyledHead>
                                <p>{title}</p>
                                <p>[{type.toUpperCase()}]</p>
                            </StyledHead>
                            <StyledBody>
                                <Button
                                    onClick={() => {
                                        history.push({ pathname: `forum/${id}` });
                                        onClick(id);
                                    }}
                                    text="View"
                                    size="small"
                                    variant="outlined"
                                    hierarchy="primary"
                                />
                                <p>{comments.length} comments</p>
                            </StyledBody>
                        </StyledWrapper>
                    );
                })}
            </StyledPostsContainer>
        </StyledContainer>
    );
};

export default ForumPreview;
