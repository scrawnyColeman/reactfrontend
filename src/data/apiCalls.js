import axios from 'axios';

const BASE_URL = 'http://localhost:8080/njoy';
const authToken = sessionStorage.getItem('authToken');
const headers = { headers: { authorization: authToken } };

/****** Course fetchrequests *******/
export const fetchAllLessons = () => axios.get(`${BASE_URL}/lessons`, headers);

export const fetchLessonById = (id) => axios.get(`${BASE_URL}/lessons/${id}`, headers);

export const fetchApprovedLessons = () => axios.get(`${BASE_URL}/lessons/approved`, headers);

export const fetchUnapprovedLessons = () => axios.get(`${BASE_URL}/lessons/unapproved`, headers);

export const fetchUserLessons = (id) => {
    return axios.get(`${BASE_URL}/user_lessons/${id}`, headers);
};

/****** Forum fetch requests *******/
export const fetchAllForumPosts = () => axios.get(`${BASE_URL}/forum_posts`, headers);

export const fetchForumPostByLesson = (id) => axios.get(`${BASE_URL}/forum_posts/lesson/${id}`, headers);

export const fetchForumPost = (id) => axios.get(`${BASE_URL}/forum_posts/${id}`, headers);

export const fetchForumPostComments = (id) => axios.get(`${BASE_URL}/comments/post/${id}`, headers);

/****** Practicals fetch requests *******/
export const fetchPracticalByLessonId = (id) => axios.get(`${BASE_URL}/practicals/lesson/${id}`, headers);

export const fetchSolutionByPracticalId = (id) => axios.get(`${BASE_URL}/solutions/practical/${id}`, headers);

/****** Quizzes fetch requests *******/
export const fetchAllQuestionsByLessonId = (id) => axios.get(`${BASE_URL}/questions/lesson/${id}`, headers);

export const fetchAllAnswersByLessonId = (id) => axios.get(`${BASE_URL}/questions/answers/lesson/${id}`, headers);

/****** Languages fetch requests *******/
export const fetchAllLanguages = () => axios.get(`${BASE_URL}/languages`, headers);

/****** Users fetch requests *******/
export const fetchUserByUsernameOrEmail = (username, email) =>
    axios.get(`${BASE_URL}/user/${username}/${email}`, headers);

/****** Users update requests *******/
export const updateUserInfo = (userId, username, email) =>
    axios.put(`${BASE_URL}/users/${userId}/username/${username}/email/${email}`, headers);

/****** Users post requests *******/
export const setUserLesson = (userId, lessonId) => axios.post(`${BASE_URL}/user/${userId}/lesson/${lessonId}`, headers);

export const createForumPost = (post) => axios.post(`${BASE_URL}/forum_posts`, post, headers);

export const createComment = (comment, userId, postId) =>
    axios.post(`${BASE_URL}/comments/${userId}/${postId}`, comment, headers);

export const deleteComment = (commentId) => axios.put(`${BASE_URL}/comments/delete/${commentId}`, headers);

export const editComment = (commentId, comment) =>
    axios.put(`${BASE_URL}/comments/edit/${commentId}`, comment, headers);

export const deleteLesson = (lessonId) => axios.delete(`${BASE_URL}/lessons/delete/${lessonId}`, headers);

export const approveLesson = (lessonId) => axios.put(`${BASE_URL}/lessons/approve/${lessonId}`, headers);

export const postLesson = (lesson, languageId) => axios.post(`${BASE_URL}/lessons/new/${languageId}`, lesson, headers);

export const postQuestions = (questions, lessonId) =>
    axios.post(`${BASE_URL}/lessons/questions/new/${lessonId}`, questions, headers);

export const postAnswers = (answers, questionId) =>
    axios.post(`${BASE_URL}/lessons/answers/new/${questionId}`, answers, headers);

export const postPractical = (practical, lessonId) =>
    axios.post(`${BASE_URL}/lessons/practical/new/${lessonId}`, practical, headers);

export const postSolution = (solution, practicalId) =>
    axios.post(`${BASE_URL}/lessons/solutions/new/${practicalId}`, solution, headers);

export const setLessonComplete = (lessonId, userId) =>
    axios.put(`${BASE_URL}/lesson/complete/${lessonId}/${userId}`, headers);
