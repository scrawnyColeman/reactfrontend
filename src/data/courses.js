import JavaIcon from '../images/java.png';
import JavaImg from '../images/javaheader.jpg';
import JSIcon from '../images/js.png';
import JSImg from '../images/jsimage.jpg';
import PythonIcon from '../images/python.png';
import PythonImg from '../images/python image.jpg';

const data = {
    courses: [
        {
            title: 'ReactJS',
            language: 'JavaScript',
            description:
                'Learn React, an open-source JavaScript library, for building fast webapps and clever User interfaces.',
            icon: JSIcon,
            header: JSImg,
            id: '1',
            activeUsers: '3514', //how many users have taken this course to generate popularity ranking
            currentUsers: ['1'],
            prereqs: [],
            theory: `yes`,
            youtubeLink: null,
        },
        {
            title: 'Lambda Functions',
            language: 'Java',
            description:
                'Produce concise lambda functions to enhance your functional programming skills. It is recommended that users complete our interfaces lesson first.',
            icon: JavaIcon,
            header: JavaImg,
            id: '2',
            activeUsers: '38',
            currentUsers: ['1', '2'],
            prereqs: ['1', '2'],
            theory: `yes`,
            youtubeLink: null,
        },
        {
            title: 'Recursion',
            language: 'JavaScript',
            description:
                'Learn recursive functions to ace all of your coding interviews. We recommend that all users complete our loops and conditionals lesson before beginning here.',
            icon: JSIcon,
            header: JSImg,
            id: '3',
            activeUsers: '802',
            currentUsers: ['0'],
            prereqs: ['1', '2'],
            theory: `A common computer programming tactic is to divide a problem into sub-problems of the same type as the original, solve those sub-problems, and combine the results. This is often referred to as the divide-and-conquer method; when combined with a lookup table that stores the results of solving sub-problems (to avoid solving them repeatedly and incurring extra computation time), it can be referred to as dynamic programming or memoization.\n
            A recursive function definition has one or more base cases, meaning input(s) for which the function produces a result trivially (without recurring), and one or more recursive cases, meaning input(s) for which the program recurs (calls itself). For example, the factorial function can be defined recursively by the equations 0! = 1 and, for all n > 0, n! = n(n − 1)!. Neither equation by itself constitutes a complete definition; the first is the base case, and the second is the recursive case. Because the base case breaks the chain of recursion, it is sometimes also called the "terminating case".\n
            The job of the recursive cases can be seen as breaking down complex inputs into simpler ones. In a properly designed recursive function, with each recursive call, the input problem must be simplified in such a way that eventually the base case must be reached. (Functions that are not intended to terminate under normal circumstances—for example, some system and server processes—are an exception to this.) Neglecting to write a base case, or testing for it incorrectly, can cause an infinite loop.\n
            For some functions (such as one that computes the series for e = 1/0! + 1/1! + 1/2! + 1/3! + ...) there is not an obvious base case implied by the input data; for these one may add a parameter (such as the number of terms to be added, in our series example) to provide a 'stopping criterion' that establishes the base case. Such an example is more naturally treated by co-recursion, where successive terms in the output are the partial sums; this can be converted to a recursion by using the indexing parameter to say "compute the nth term (nth partial sum)" `,
            youtubeLink: `https://www.youtube.com/watch?v=Mv9NEXX1VHc`,
        },
        {
            title: 'Polymorphism',
            language: 'Java',
            description:
                'Meaning to take many forms, polymorphism is among the four pillars of Object-oriented programming; a must know for all programmers.',
            icon: JavaIcon,
            header: JavaImg,
            id: '4',
            activeUsers: '604',
            currentUsers: ['1', '2'],
            prereqs: ['1', '2'],
            theory: `The word polymorphism is used in various contexts and describes situations in which something occurs in several different forms. In computer science, it describes the concept that objects of different types can be accessed through the same interface. Each type can provide its own, independent implementation of this interface. It is one of the core concepts of object-oriented programming (OOP). \nIf you’re wondering if an object is polymorphic, you can perform a simple test. If the object successfully passes multiple is-a or instanceof tests, it’s polymorphic. As I’ve described in my post about inheritance, all Java classes extend the class Object. Due to this, all objects in Java are polymorphic because they pass at least two instanceof checks.`,
            youtubeLink: `https://www.youtube.com/watch?v=tIWm3I_Zu7I`,
        },
        {
            title: 'Control Flow and Conditionals',
            language: 'Python',
            description:
                'Learn how to write a program that is able to do hundreds of hours of human cognition in seconds through repetition and scripted decision making',
            icon: PythonIcon,
            header: PythonImg,
            id: '5',
            activeUsers: '190',
            currentUsers: ['0'],
            prereqs: ['1', '2'],
            theory: `yes`,
            youtubeLink: null,
        },
        {
            title: 'Classes and Objects',
            language: 'Python',
            description: 'Learn the fundamental elements that make up the world of Object-Oriented Programming.',
            icon: PythonIcon,
            header: PythonImg,
            id: '6',
            activeUsers: '334',
            currentUsers: ['0'],
            prereqs: ['1', '2'],
            theory: `yes`,
            youtubeLink: null,
        },
        {
            title: 'ES6 Arrow Functions',
            language: 'JavaScript',
            description:
                'Learn arrow functions - introduced with the ECMAScript 6 - and discover how to write shorter, more concise functions with more capable than before.',
            icon: JSIcon,
            header: JSImg,
            id: '7',
            activeUsers: '2154',
            currentUsers: ['0'],
            prereqs: ['1', '2'],
            theory: `yes`,
            youtubeLink: null,
        },
    ],
};

export default data;
