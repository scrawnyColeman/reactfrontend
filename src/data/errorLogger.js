export const errorLogger = (err) => {
    console.log(err);
};
