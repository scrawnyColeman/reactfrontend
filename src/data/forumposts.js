import courseData from './courses';

var key = 1;
const data = {
    posts: [
        {
            title: 'ArrayLists vs Linked Lists',
            type: 'HELP',
            question:
                'So I understand that for different situations, either the ArrayList API or the LinkedList API will be more efficient. So what should I look for in terms of design that will tell me which one of those structures is recommended? Is it more related to preference rather than specific uses?',
            id: `${key++}`,
            comments: [
                {
                    commentid: '1',
                    comment: "No it can't my guy",
                    parentid: '0',
                    authorusername: 'anon',
                },
                {
                    commentid: '2',
                    comment: 'Oh okay. That sucks',
                    parentid: '1',
                },
                {
                    commentid: '3',
                    comment: 'Helpful post. Good Job... OP, you can do it by...',
                    parentid: '1',
                },
                {
                    commentid: '4',
                    comment: 'Thank you!!',
                    parentid: '3',
                },
            ],
        },
        {
            title: 'Can this be done in one loop?',
            type: 'HELP',
            question: `Hey all, I'm currently working my way through the MOOC course and I've just completed an exercise where a scanner asks for a list of numbers from the user, and then it prints out what the smallest number in the index is, and prints out every index it is at. I'm wondering if this could be completed in just one loop.`,
            id: `${key++}`,
            comments: [
                {
                    commentid: '5',
                    comment: "No it can't my guy",
                    parentid: '0',
                },
                {
                    commentid: '6',
                    comment: 'Oh okay. That sucks',
                    parentid: '5',
                },
                {
                    commentid: '7',
                    comment: 'Helpful post. Good Job smh... OP, you can do it by...',
                    parentid: '5',
                },
                {
                    commentid: '8',
                    comment: 'Thank you!!',
                    parentid: '7',
                },
            ],
        },
    ],
};

courseData.courses.forEach((course) => {
    data.posts.push({
        title: `${course.title} [${course.language}]`,
        id: `${key++}`,
        type: 'LESSON',
        question: `This thread is for discussing the ${course.title} lesson, taught in ${course.language}`,
        comments: [],
    });
});
export default data;
