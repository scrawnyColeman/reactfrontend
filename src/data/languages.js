export const data = {
    languages: [
        {
            id: '1',
            name: 'Java',
        },
        {
            id: '2',
            name: 'JavaScript',
        },
        {
            id: '3',
            name: 'Python',
        },
    ],
};
