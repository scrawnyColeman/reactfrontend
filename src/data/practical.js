import axios from 'axios';

const BASE_URL = `http://localhost:8080/njoy`;
export const fetchPracticalByLessonId = (id) => axios.get(`${BASE_URL}/practicals/lesson/${id}`);
export const fetchSolutionByPracticalId = (id) => axios.get(`${BASE_URL}/solutions/practical/${id}`);

export default fetchPracticalByLessonId;
