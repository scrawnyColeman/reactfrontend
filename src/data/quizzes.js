import axios from 'axios';

const BASE_URL = `http://localhost:8080/njoy`;

export const fetchAllQuestionsByLessonId = (id) => axios.get(`${BASE_URL}/questions/lesson/${id}`);
export const fetchAllAnswersByLessonId = (id) => axios.get(`${BASE_URL}/questions/answers/lesson/${id}`);

export default fetchAllAnswersByLessonId;
