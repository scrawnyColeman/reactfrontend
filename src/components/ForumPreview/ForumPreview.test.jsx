import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import ForumPreview from './component';
import data from '../../data/forumposts.js';

Enzyme.configure({ adapter: new Adapter() });

describe('ForumPreview component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<ForumPreview posts={data.posts} />);
        expect(wrapper).toMatchSnapshot();
    });
});
