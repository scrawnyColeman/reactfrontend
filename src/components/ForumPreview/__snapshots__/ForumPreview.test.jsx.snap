// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForumPreview component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    Forum Posts:
  </styled.div>
  <styled.div>
    <styled.div
      key="1"
    >
      <styled.div>
        <p>
          ArrayLists vs Linked Lists
        </p>
        <p>
          [
          HELP
          ]
        </p>
      </styled.div>
      <styled.div>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="View"
          variant="outlined"
        />
        <p>
          4
           comments
        </p>
      </styled.div>
    </styled.div>
    <styled.div
      key="2"
    >
      <styled.div>
        <p>
          Can this be done in one loop?
        </p>
        <p>
          [
          HELP
          ]
        </p>
      </styled.div>
      <styled.div>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="View"
          variant="outlined"
        />
        <p>
          4
           comments
        </p>
      </styled.div>
    </styled.div>
    <styled.div
      key="3"
    >
      <styled.div>
        <p>
          ReactJS [JavaScript]
        </p>
        <p>
          [
          LESSON
          ]
        </p>
      </styled.div>
      <styled.div>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="View"
          variant="outlined"
        />
        <p>
          0
           comments
        </p>
      </styled.div>
    </styled.div>
    <styled.div
      key="4"
    >
      <styled.div>
        <p>
          Lambda Functions [Java]
        </p>
        <p>
          [
          LESSON
          ]
        </p>
      </styled.div>
      <styled.div>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="View"
          variant="outlined"
        />
        <p>
          0
           comments
        </p>
      </styled.div>
    </styled.div>
    <styled.div
      key="5"
    >
      <styled.div>
        <p>
          Recursion [Python]
        </p>
        <p>
          [
          LESSON
          ]
        </p>
      </styled.div>
      <styled.div>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="View"
          variant="outlined"
        />
        <p>
          0
           comments
        </p>
      </styled.div>
    </styled.div>
    <styled.div
      key="6"
    >
      <styled.div>
        <p>
          Polymorphism [Java]
        </p>
        <p>
          [
          LESSON
          ]
        </p>
      </styled.div>
      <styled.div>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="View"
          variant="outlined"
        />
        <p>
          0
           comments
        </p>
      </styled.div>
    </styled.div>
    <styled.div
      key="7"
    >
      <styled.div>
        <p>
          Control Flow and Conditionals [Python]
        </p>
        <p>
          [
          LESSON
          ]
        </p>
      </styled.div>
      <styled.div>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="View"
          variant="outlined"
        />
        <p>
          0
           comments
        </p>
      </styled.div>
    </styled.div>
    <styled.div
      key="8"
    >
      <styled.div>
        <p>
          Classes and Objects [Python]
        </p>
        <p>
          [
          LESSON
          ]
        </p>
      </styled.div>
      <styled.div>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="View"
          variant="outlined"
        />
        <p>
          0
           comments
        </p>
      </styled.div>
    </styled.div>
    <styled.div
      key="9"
    >
      <styled.div>
        <p>
          ES6 Arrow Functions [JavaScript]
        </p>
        <p>
          [
          LESSON
          ]
        </p>
      </styled.div>
      <styled.div>
        <Button
          hierarchy="primary"
          onClick={[Function]}
          size="small"
          text="View"
          variant="outlined"
        />
        <p>
          0
           comments
        </p>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
